### DISCUSSION TITLE
Are PAW projectors used in .spn file?

### DISCUSSION THREAD

**Discussion by stepan-tsirkin** [Chengcheng-Xiao/VASP2WAN90_v2_fix#19] [General]
Hi @Chengcheng-Xiao 

Just one question to  make sure my understanding is correct. In your patch, to calculate the spn file is the full PAW wavefunction is used?  Or only the plane-wave part - the one given by plane-wave coeficients in the WAVECAR file?

In turn, my utility [vaspspn](https://docs.wannier-berri.org/en/master/docs/vaspspn.html) uses only the plane-wave part (written in the WAVECAR) and normalizes it. This is a rough approximation but it used to work qualitatively. 

[Here](https://github.com/wannier-berri/wannier-berri/issues/274)  @ma455173220  reported wrong results with vaspspn, but good results with your patch. So I want to find out if it is a bug that somehow crawled into vaspspn over time, or is it the issue of using only plane-wave part.

Best,
Stepan.

**Comment by Chengcheng-Xiao** [comment]
Hello again @stepan-tsirkin!

To answer your question, my patch calculates the `.spn` file using the full PAW formalism, including the augmented part (the projector stuff you mentioned).

> my utility [vaspspn](https://docs.wannier-berri.org/en/master/docs/vaspspn.html) uses only the plane-wave part (written in the WAVECAR) and normalizes it.

I am actually not sure if normalizing the pseudo-wavefunction directly is a good idea. In VASP, the correct moments are recovered by adding the "compensation charges" using the multipole expansion method (see https://journals.aps.org/prb/pdf/10.1103/PhysRevB.59.1758). Directly normalizing the wavefunctions might be problematic. Regrettably, off the top of my head, I don't really have a good idea of calculating the `.spn` file with only the pseudo-wavefunctions and no information from the PAW dataset.

Best,
Chengcheng

**Reply by stepan-tsirkin** [reply]
Thanks  @Chengcheng-Xiao ,

> I am actually not sure if normalizing the pseudo-wavefunction directly is a good idea.

It is a **bad** idea in general. It only makes sense in the limit where a spin state is a pure spin state. Say, for some wavefunciton $\psi$ only the "spin-up" component of the spinor is non-zero., then we will get $\langle \psi| \sigma_z | \psi>==1$  both in case we use afull PAW wavefunction  or a normalized pseudo-wavefunction. 

But for off-diagonal matrix elements, I guess, it is strongly wrong.
